lib/jwt/decode.rb

```diff
@@ -65,7 +71,7 @@ def allowed_algorithms
 
     def find_key(&keyfinder)
       key = (keyfinder.arity == 2 ? yield(header, payload) : yield(header))
-      raise JWT::DecodeError, 'No verification key available' unless key
+      raise JWT::DecodeError, 'No verification key available' unless key && key.empty?
```

Comment:
Think this is assuming the key returned from the block is an array. So if a single string is returned i think this will fail with a `NoMethodError`

Reply:
I checked in ruby 2.4.7+. `String#empty?` and `Array#empty?` are both valid methods. Am I missing something obvious here? 🤔

Reply:
You're right. This works just fine.